### PR TITLE
Fix Skip task confirmation dialog not showing properly

### PR DESCRIPTION
**Pull request by JSunde**
Also included a couple UX tweaks for the confirmation dialog

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt [hunk 1]
```diff
@@ -125,14 +126,14 @@ abstract class AbstractTaskFragment<T : AbstractTaskViewModel> : AbstractFragmen
       .showIfTrue(viewModel.isTaskOptional())
 
   private fun onSkip() {
-    if (viewModel.responseText.value.isNullOrEmpty()) {
+    if (viewModel.taskData.value?.orElseGet { null }?.isNullOrEmpty() != false) {
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt:129]: Can we use `taskDataFlow` instead? `taskData` uses LiveData which we are trying to slowly phase out.

File: ground/src/test/java/com/google/android/ground/ui/datacollection/tasks/multiplechoice/MultipleChoiceTaskFragmentTest.kt [hunk 2]
```diff
@@ -149,6 +154,29 @@ class MultipleChoiceTaskFragmentTest :
...
+  fun testActionButtons_dataEntered_skipButtonTapped_confirmationDialogIsShown() {
+    val multipleChoice = MultipleChoice(options, MultipleChoice.Cardinality.SELECT_ONE)
+    setupTaskFragment<MultipleChoiceTaskFragment>(task.copy(multipleChoice = multipleChoice))
+
+    onView(withText("Option 1")).perform(click())
+
+    onView(withText("Skip")).perform(click())
+    onView(withText(context.getString(R.string.skip_dialog_title)))
+      .inRoot(isDialog())
+      .check(matches(isDisplayed()))
```
Comment (shobhitagarwal1612) [inline ground/src/test/java/com/google/android/ground/ui/datacollection/tasks/multiplechoice/MultipleChoiceTaskFragmentTest.kt:167]: Robolectric has a utility [`ShadowDialog`](https://robolectric.org/javadoc/4.3/org/robolectric/shadows/ShadowDialog.html) that can be used here. 

e.g. ShadowDialog.getLatestDialog(), ShadowDialog.getShownDialogs()

File: ground/src/test/java/com/google/android/ground/ui/datacollection/tasks/multiplechoice/MultipleChoiceTaskFragmentTest.kt [hunk 3]
```diff
@@ -149,6 +154,29 @@ class MultipleChoiceTaskFragmentTest :
...
+  }
+
+  @Test
+  fun testActionButtons_noDataEntered_skipButtonTapped_confirmationDialogIsNotShown() {
+    val multipleChoice = MultipleChoice(options, MultipleChoice.Cardinality.SELECT_ONE)
+    setupTaskFragment<MultipleChoiceTaskFragment>(task.copy(multipleChoice = multipleChoice))
+
+    onView(withText("Skip")).perform(click())
+    onView(withText(context.getString(R.string.skip_dialog_title)))
+      .inRoot(isDialog()).check(doesNotExist())
```
Comment (shobhitagarwal1612) [inline ground/src/test/java/com/google/android/ground/ui/datacollection/tasks/multiplechoice/MultipleChoiceTaskFragmentTest.kt:177]: Same as above

**Review by shobhitagarwal1612 — CHANGES_REQUESTED** [review]

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt (hunk 1 above)
Comment (JSunde) [reply ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt:129]: I tried using the taskDataFlow, but for some reason I couldn't get it to work. I don't know if there's some issue with the flow not being updated properly or what.

Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt:129]: You can use `taskDataFlow.value != null` to check whether data is present or not.

File: ground/src/test/java/com/google/android/ground/ui/datacollection/tasks/multiplechoice/MultipleChoiceTaskFragmentTest.kt [hunk 4]
```diff
@@ -149,6 +158,26 @@ class MultipleChoiceTaskFragmentTest :
...
+    assertThat(ShadowAlertDialog.getLatestDialog().isShowing).isTrue()
+  }
+
+  @Test
+  fun testActionButtons_noDataEntered_skipButtonTapped_confirmationDialogIsNotShown() {
+    val multipleChoice = MultipleChoice(options, MultipleChoice.Cardinality.SELECT_ONE)
+    setupTaskFragment<MultipleChoiceTaskFragment>(task.copy(multipleChoice = multipleChoice))
+
+    onView(withText("Skip")).perform(click())
+    assertThat(ShadowAlertDialog.getLatestDialog().isShowing).isFalse()
```
Comment (shobhitagarwal1612) [inline ground/src/test/java/com/google/android/ground/ui/datacollection/tasks/multiplechoice/MultipleChoiceTaskFragmentTest.kt:178]: `getLatestDialog()` should return null if the dialog is never created.

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt (hunk 1 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt:129]: It'd be nice to add a helper method to the viewModel to add some level of abstraction.

```
fun hasData(): Boolean = taskDataFlow.value != null
```

**Pull request merged**